src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -52,25 +54,35 @@ template Batch
             (instructionCid, instruction) <- fetchByKey @Instruction (requestors, instructionId)
             s <- case instruction.allocation of
               PLEDGE senderTransferableCid -> do
-                let holdingCid : ContractId Base.I = coerceContractId senderTransferableCid
-                holding <- fetch holdingCid
+                holding <- fetch $ toInterfaceContractId @Base.I senderTransferableCid
                 fetchAccount holding
-                pure [(getAccount holding, instructionId)]
-              _ -> pure []
+                pure $ Some $ (getAccount holding, instructionId)
```

Comment:
```suggestion
                pure . Some $ (getAccount holding, instructionId)
```

Reply:
done

Reply:
just fyi - you could have pressed the "Commit suggestion" button to make the change :)